### DISCUSSION TITLE
Hi, how can I fork using the Git command tool?

### DISCUSSION THREAD

**Discussion by ysrazsingh** [community/community#112997] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hi, how can I fork using the Git command tool?

For example, I cloned a public repository where my origin is set to git@github.com:chetan-deshpande2/marketplace-BE.git. Then, I created a repository in my GitHub account and added origin with the name 'backup' using git remote add backup git@github.com:YsrajSingh/c-t.git.

Now, I want to push all branches and commits, just like a fork. So, I used git push -u backup --mirror, but this only pushed the main branch with all commits. I want to push all branches and commits which are present in git@github.com:chetan-deshpande2/marketplace-BE.git.

**Comment by SebastienXia** [comment]
Hello @YsrajSingh 

I let you read this small blog about the fork in Git command tool : https://www.theserverside.com/blog/Coffee-Talk-Java-News-Stories-and-Opinions/command-line-GitHub-fork-CLI-terminal-shell

enjoy :)

**Reply by ysrazsingh** [reply]
Thank You for your response. But here is answer of what I want (you can found this below also)
https://github.com/orgs/community/discussions/112997#discussioncomment-8821224

**Comment by fares-ahmedd** [comment]
To fork a repository using the Git command line, you can follow these steps:

Navigate to GitHub: Go to the GitHub repository you want to fork in your web browser.

Find Fork Button: Click on the "Fork" button at the top-right corner of the repository page. This will create a copy of the repository under your GitHub account.

Clone Forked Repository: Once you have forked the repository, you need to clone it to your local machine. Open your terminal or command prompt and navigate to the directory where you want to clone the repository.

Clone Command: Use the git clone command followed by the URL of your forked repository. You can find the URL by clicking on the green "Code" button on your repository page and copying the HTTPS or SSH URL.

**Reply by ysrazsingh** [reply]
Thank You for your response. But here is answer of what I want (you can found this below also)
https://github.com/orgs/community/discussions/112997#discussioncomment-8821224

**Comment by stjepanDeveloper** [comment, marked as answer]
I think your git push -u backup --mirror command is fine, as it should push all branches and commits.
Try to check out if you cloned a perfect clone, so that all the branches and commits are included.
You can use the command line like this to make a complete bare clone :

git clone --bare https://yourlinkhere.git

You can verify that you have all the branches before pushing with :
git branch -a

And list tags :
git tag

Try pushing the complete clone with the same command as before and see if it includes all the branches.

**Reply by ysrazsingh** [reply]
Thank you very much; the --bare option works for me. The rest of the commands I used are the same. Can you please help me understand what --bare does here and let me know about other Git tools that developers must know?

Should we use all commands like pull, push, checkout in a bare repo as well? What is the difference between a clone created by --bare and one created normally?

**Reply by ysrazsingh** [reply]
@stjepanDeveloper

**Reply by stjepanDeveloper** [reply]
What the --bare command basically does it creates a copy of the repository's history and files as they exist in the .git directory, but without a working directory. 
You have a bare and non bare type of repositories. A bare repository is essentially the contents of the .git directory of a standard (non-bare) Git repository, without a working directory. 
In a non-bare repository, the .git directory contains all the repository metadata, configuration, and repository database, and the working directory contains your actual files. 
Bare repositories are often used on servers to host the central copy of a repository where users push and pull changes. They are ideal for collaboration among developers and are the backbone of remote repositories on services like GitHub, GitLab, and Bitbucket.
In summary, git clone --bare is a specialized command used for creating an exact, full-fidelity copy of a repository's Git database, suitable for use as a central repository or for mirroring purposes. It’s an essential tool for Git administrators and developers working in collaborative environments or managing multiple repository copies.

As for the other essential commands there are a lot of them so you should check the documentation for that. But at the top of my mind here are some :
git commit, git add, git status, git branch, git merge, git checkout, git push&pull, git fetch, git log, git stash and git clean...

**Reply by ysrazsingh** [reply]
Great! Thanks for getting back to me.
Today, I learned something new about Git! Even though I've been using it for over a year, I mostly relied on commands like git add, git commit, git push, etc.  (only basic commands)

I never really dug into the documentation before (like you suggested), but now I understand the importance of doing so for a deeper understanding.
This new knowledge motivates me to explore the documentation part of each and gain a deeper understanding of the what I've been using.

**Reply by stjepanDeveloper** [reply]
No problem, good luck with your programming!!
